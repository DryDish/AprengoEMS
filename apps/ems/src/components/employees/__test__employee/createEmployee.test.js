import { render, cleanup } from '@testing-library/react';
import {act, renderHook} from '@testing-library/react-hooks'
import React from 'react';
import ReactDOM from 'react-dom'; 

import '@testing-library/jest-dom/extend-expect.js';
import renderer from 'react-test-renderer';
import CreateEmployee from '../createEmployee';

afterEach(cleanup); 
// Cleans up after each test. If this is not added
// it runs all tests at the same time. This can create
// several conflicts when it renders multiple components at once 

it('renders without creashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<CreateEmployee />, div);
}) 

it('matches snapshot', () => {
    const tree = renderer.create(<CreateEmployee />).toJSON(); 
    expect(tree).toMatchSnapshot()
})

it('renders createEmployee correctly', () => {
    const { getByTestId } = render(<CreateEmployee />);
    expect(getByTestId("createEmployee"))
        .toHaveTextContent("Create Employee");
}) 

// describe('input', () => 
// {
//     it('Converts hours into minutes properly', () => 
//     {
//         const {result} = renderHook(CreateEmployee) 

//         act(() => 
//         {
//             result.current.updateInfo('minutesWorked', 3.5)
//         })

//         expect(result.current.firstName).toBe(210)
//     })
// })

