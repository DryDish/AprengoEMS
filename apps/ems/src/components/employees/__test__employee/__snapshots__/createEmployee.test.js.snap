// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="createPage"
  data-testid="createEmployee"
>
  <div
    className="container"
  >
    <div
      className="row"
      style={
        Object {
          "paddingTop": "7%",
        }
      }
    >
      <div
        className="col-lg-3 col-md-2"
      />
      <div
        className="col-lg-6 col-md-8 col-12"
      >
        <h1
          style={
            Object {
              "color": "black",
            }
          }
        >
          Add Employee
        </h1>
        <br />
        <div
          className="md-form"
        >
          
          <input
            aria-disabled={false}
            className="form-control"
            data-test="input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            type="text"
            value=""
          />
          <label
            aria-labelledby=""
            className=""
            data-error=""
            data-success=""
            id=""
            onClick={[Function]}
          >
            First Name
          </label>
        </div>
        <div
          className="md-form"
        >
          
          <input
            aria-disabled={false}
            className="form-control"
            data-test="input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            type="text"
            value=""
          />
          <label
            aria-labelledby=""
            className=""
            data-error=""
            data-success=""
            id=""
            onClick={[Function]}
          >
            Last Name
          </label>
        </div>
        <div
          className="md-form"
        >
          
          <input
            aria-disabled={false}
            className="form-control"
            data-test="input"
            disabled={false}
            id="phoneInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            type="number"
            value=""
          />
          <label
            aria-labelledby=""
            className=""
            data-error=""
            data-success=""
            htmlFor="phoneInput"
            id=""
            onClick={[Function]}
          >
            Phone Number
          </label>
        </div>
        <div
          className="md-form"
        >
          
          <input
            aria-disabled={false}
            className="form-control"
            data-test="input"
            data-testid="getHoursWorkedInMinutes"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            step="0.25"
            type="number"
            value=""
          />
          <label
            aria-labelledby=""
            className=""
            data-error=""
            data-success=""
            id=""
            onClick={[Function]}
          >
            Weekly Workhours
          </label>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn-dark btn Ripple-parent"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Create Employee
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
